### PR TITLE
Improve apiUsageMonitoring filter logging

### PR DESCRIPTION
**Pull request by RomanZavodskikh**
Updates #2186

Signed-off-by: Roman Zavodskikh <roman.zavodskikh@zalando.de>

### CODE REVIEW

**Comment by RomanZavodskikh** [comment]
I tested skipper initializing with such route:
```
baidu:
        Path("/baidu")
        -> apiUsageMonitoring(`{
            "application_id": "pop-proxy",
            "api_id": "bbf18a86-1cfb-11eb-adc1-0242ac120003",
            "path_templates": [
                "solr/customer/select",
                "solr/customer/get",
                "solr/customer/update",
                "solr/order_position/select",
                "solr/order_position/get",
                "solr/order_position/update",
                "solr/order/select",
                "solr/order/get",
                "solr/order/update",
                "fds/solr/customer/select",
                "fds/solr/customer/get",
                "fds/solr/customer/update",
                "fds/solr/order_position/select",
                "fds/solr/order_position/get",
                "fds/solr/order_position/update",
                "fds/solr/order/select",
                "fds/solr/order/get",
                "fds/solr/order/update"
            ]
        }`, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0)
        -> setRequestHeader("Host", "www.baidu.com")
        -> setPath("/s")
        -> setQuery("wd", "godoc skipper")
        -> "http://www.baidu.com";
```

Without this PR logs are the following:
```
[APP]INFO[0000] route settings received                      
[APP]ERRO[0000] args[1] ignored: expecting a string, was %!t(float64=0)  filter=apiUsageMonitoring
[APP]ERRO[0000] args[2] ignored: expecting a string, was %!t(float64=0)  filter=apiUsageMonitoring
[APP]ERRO[0000] args[3] ignored: expecting a string, was %!t(float64=0)  filter=apiUsageMonitoring
[APP]ERRO[0000] args[4] ignored: expecting a string, was %!t(float64=0)  filter=apiUsageMonitoring
[APP]ERRO[0000] args[5] ignored: expecting a string, was %!t(float64=0)  filter=apiUsageMonitoring
[APP]ERRO[0000] args[6] ignored: expecting a string, was %!t(float64=0)  filter=apiUsageMonitoring
[APP]ERRO[0000] args[7] ignored: expecting a string, was %!t(float64=0)  filter=apiUsageMonitoring
[APP]ERRO[0000] args[8] ignored: expecting a string, was %!t(float64=0)  filter=apiUsageMonitoring
[APP]ERRO[0000] args[9] ignored: expecting a string, was %!t(float64=0)  filter=apiUsageMonitoring
[APP]ERRO[0000] args[10] ignored: expecting a string, was %!t(float64=0)  filter=apiUsageMonitoring
[APP]ERRO[0000] args[11] ignored: expecting a string, was %!t(float64=0)  filter=apiUsageMonitoring
[APP]INFO[0000] route settings applied 
```

With this PR is the following:
```
[APP]INFO[0000] route settings received                      
[APP]ERRO[0000] args[1] ignored: expecting a string, was %!t(float64=0)  filter=apiUsageMonitoring
[APP]ERRO[0000] args[2] ignored: expecting a string, was %!t(float64=0)  filter=apiUsageMonitoring
[APP]ERRO[0000] args[3] ignored: expecting a string, was %!t(float64=0)  filter=apiUsageMonitoring
[APP]INFO[0000] route settings applied   
```

So from my side rate.Sometimes seems working. :-)

File: filters/apiusagemonitoring/spec.go [hunk 1]
```diff
@@ -120,6 +123,31 @@ type apiUsageMonitoringSpec struct {
...
 	realmsTrackingMatcher *regexp.Regexp
 	unknownPath           *pathInfo
+	onPanicSometimes      rate.Sometimes
+}
+
+func (s *apiUsageMonitoringSpec) sometimesErrorf(args ...interface{}) {
+	s.onPanicSometimes.Do(func() {
+		log.Errorf(args[0].(string), args[1:]...)
+	})
+}
```
Comment (AlexanderYastrebov) [inline filters/apiusagemonitoring/spec.go:133]: I think it makes sense to check log.Errorf signature and use the same for this function.
Then I propose we name them simply `errorf`, `warnf` and `infof`

File: docs/reference/filters.md [hunk 2]
```diff
@@ -2374,12 +2374,12 @@ Those metrics are not identifying the realm and client. They always have `*` in
 Example:
 
 ```
-                                                                                     + Realm
-                                                                                     |
+                                                                                      + Realm
```
Comment (AlexanderYastrebov) [inline docs/reference/filters.md:2377]: What this change is for?

Comment (RomanZavodskikh) [reply docs/reference/filters.md:2377]: To make the annotations in doc point to the right place.
Before:
Screenshot from 2023-03-06 17-40-05
After:
Screenshot from 2023-03-06 17-40-11

Comment (AlexanderYastrebov) [reply docs/reference/filters.md:2377]: It looks ok to me 

https://github.com/zalando/skipper/blob/master/docs/reference/filters.md#available-metrics

Screenshot from 2023-03-06 17-48-55

Comment (AlexanderYastrebov) [reply docs/reference/filters.md:2377]: Ah, right, should point to the star

**Comment by RomanZavodskikh** [comment]
I think I need to create unit tests like in `proxy/proxy_test.go`.

**Comment by AlexanderYastrebov** [comment]
:+1:

**Comment by AlexanderYastrebov** [comment]
We do not usually test logging output so it is fine to have this change like it is

**Comment by RomanZavodskikh** [comment]
:

**Comment by RomanZavodskikh** [comment]
:+1:

**Pull request merged**